config/ulib_config.h

```diff
@@ -53,6 +53,29 @@ extern "C" {
  */
 #define AZ_ULIB_CONFIG_MAX_LOG_SIZE        256
 
+/**
+ * @brief   IPC public API shall validate the contract
+ *
+ *  This definition enables the argument check on all IPC public APIs. This check will verify if the
+ *      provided argument follow the contract defined in the documentation. 
+ *  Commenting this definition, the IPC public APIs will not test any of the received arguments.
+ */
+#define AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT
```

Comment:
>AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT [](start = 8, length = 36)

I'm considering to add one key exclusively to validate the contract on the `ipc_call`, because this is the critical path on IPC and any extra test may have a huge impact on the performance.

Any thoughts?

Reply:
so this would effectively be leaving out which function calls to not check?

---
In reply to: [338166167](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338166167) [](ancestors = 338166167)

Reply:
No, the new one will be used on the `az_ulib_ipc_call`, and it will change the behavior for all calls.

---
In reply to: [338320767](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338320767) [](ancestors = 338320767,338166167)

Reply:
It will replace `AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT` here...

```
static inline AZ_ULIB_RESULT az_ulib_ipc_call(
    az_ulib_ipc_interface_handle interface_handle,
    az_ulib_action_handle method_handle,
    const void *const model_in,
    const void *model_out) {
#ifdef AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT
  return _az_ulib_ipc_call(
      (_az_ulib_ipc_interface_handle)interface_handle, method_handle, model_in, model_out);
#else
  return _az_ulib_ipc_call_no_test(
      (_az_ulib_ipc_interface_handle)interface_handle, method_handle, model_in, model_out);
#endif /* AZ_ULIB_CONFIG_IPC_VALIDATE_CONTRACT */
}
```

---
In reply to: [338660790](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338660790) [](ancestors = 338660790,338320767,338166167)

Reply:
After chat with Dane, we concluded that, for now, we should keep only one key. If there is customers complaining about it, we can change in the future.

---
In reply to: [338661261](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338661261) [](ancestors = 338661261,338660790,338320767,338166167)